### PR TITLE
Fix android getAll

### PR DESCRIPTION
**Pull request by morenoh149**
* fix error 14 on real device
* peform query only once against [Contactables table](http://developer.android.com/reference/android/provider/ContactsContract.CommonDataKinds.Contactables.html)
* differs from iOS api in that the entire display name is passed into the `givenName` field

### CODE REVIEW

**Comment by morenoh149** [comment]
splitting the displayName by spaces is brittle. I'm tempted to simply return displayName in the `givenName` field and document it to users.

**Comment by morenoh149** [comment]
gonna look into using http://developer.android.com/reference/android/provider/ContactsContract.ContactNameColumns.html#DISPLAY_NAME_SOURCE and http://developer.android.com/reference/android/provider/ContactsContract.CommonDataKinds.StructuredName.html to see if I can get the displayName decomposed

**Comment by morenoh149** [comment]
@rt2zz I tried to query the name table as I show here http://stackoverflow.com/questions/4301064/how-to-get-the-first-name-and-last-name-from-android-contacts/34478417#34478417 I can't get the `given_name` to show from contacts on my phone.

Would you be interested in making the ios api conform to the android api? otherwise we can just merge this with the readme note.

In other good news though, I was able to avoid the number requirement by filtering by `IN_VISIBLE_GROUP` field.

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java [hunk 1]
```diff
@@ -1,157 +1,167 @@
...
+  /*
+   * Returns all contactable records on phone
+   * queries CommonDataKinds.Contactables to get phones and emails
+   */
   @ReactMethod
   public void getAll(Callback callback) {
     ContentResolver cr = getReactApplicationContext().getContentResolver();
-    Cursor cur = cr.query(ContactsContract.Contacts.CONTENT_URI, null, null, null, null);
+    Uri uri = CommonDataKinds.Contactables.CONTENT_URI;
+    String selection = ContactsContract.Contacts.IN_VISIBLE_GROUP + " = " + 1;
```
Comment (morenoh149) [inline android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]: make `selection` null if you want all contacts @tUrG0n

**Comment by morenoh149** [comment]
@rt2zz anything blocking this?

**Comment by NickyYo** [comment]
Thanks for this @morenoh149 ! just tried this fix, a lot faster!

**Comment by tUrG0n** [comment]
@rt2zz yup indeed its waayyy faster. Do merge the pull request :)

**Comment by rt2zz** [comment]
@morenoh149 awesome! I have not tested this myself but thanks @tUrG0n and @NickyYo for testing. I will take a deeper look into this next week.

**Comment by niftylettuce** [comment]
@rt2zz @morenoh149 Why can't you just do a Java version of JavaScript's `.split(' ')[0]` and `.split(' ')[1]` for <https://github.com/rt2zz/react-native-contacts/commit/d0ebd3230abd57087798f89a049299e989012ed9#diff-4007ac2f1fac9bbff88a652279d4ec7aR76>?  That way the `givenName` will be correct as would the other two.

**Comment by morenoh149** [comment]
It's brittle. It breaks down on names such as `Carmen de la Cruz`. That approach would return ['Carmen', 'de', 'la Cruz'] even if the user originally inputted ['Carmen', 'de la Cruz'].

**Comment by niftylettuce** [comment]
@morenoh149 No, just make it so it returns `Carmen` as first and as last `de la Cruz`, first should be the first split, second should be the rest after the first split.

**Comment by niftylettuce** [comment]
@morenoh149 Equivalent of `firstName = givenName.split(' ')[0]` and then you get indexOf whitespace... so `var whitespace = givenName.indexOf(' ')` and then `lastName = givenName.substr(whitespace);`

**Comment by niftylettuce** [comment]
This would make it like iOS I'd think, right?

**Comment by morenoh149** [comment]
no middlename?

**Comment by niftylettuce** [comment]
I think that we can make a note in the Readme that middle names are not parsed yet due to this issue.

**Comment by niftylettuce** [comment]
Have you looked into how Apache Cordova parses middle names?

**Comment by tUrG0n** [comment]
@rt2zz is the latest version bumped on npm? `package.json` is still at `0.1.5`

**Comment by rt2zz** [comment]
@tUrG0n waiting to bump the version as there are a lot of changes going on right now. In the meantime you can get the latest by installing from git master: `npm install rt2zz/react-native-contacts`

**Comment by tUrG0n** [comment]
@rt2zz  sure :)

**Pull request merged**